### ISSUE TITLE
Symbol(s) not found for architecture arm64

### ISSUE THREAD

**Issue by FrankdenUijl** [yasirkula/UnityNativeGallery#236]
**Unable make a iOS build in Unity Cloud Build**

I have the following issue in Unity Cloud Build:
![image](https://user-images.githubusercontent.com/8240783/144276072-a0b42bfd-1d22-49e1-ad7c-11be23f009f9.png)
![image](https://user-images.githubusercontent.com/8240783/144276151-193f18e2-4c25-4b0a-a0a4-e0e8bb6611fe.png)

[craytive-technologies-bv-baselinez-ios-baselinez-dev-40-BaselineZ-Dev-Unity-iPhone.log](https://github.com/yasirkula/UnityNativeGallery/files/7635689/craytive-technologies-bv-baselinez-ios-baselinez-dev-40-BaselineZ-Dev-Unity-iPhone.log)

**Reproduction steps**

Include the package, run it in Unity Cloud Build

- Unity version: 2020.3.21f1
- Platform: iOS
- How did you download the plugin: GitHub

**Comment by FrankdenUijl** [comment]
![image](https://user-images.githubusercontent.com/8240783/144279438-883458e4-2834-4348-bef2-99bf50245e33.png)
I use automated setup.

**Comment by yasirkula** [comment]
Are there any editor scripts that call "SetBuildProperty" with "OTHER_LDFLAGS" parameter in the entirety of the project?

**Comment by FrankdenUijl** [comment]
We have the following:
project.SetBuildProperty(targetGUID, "ARCHS", "arm64");
proj.SetBuildProperty (targetGUID, "IPHONEOS_DEPLOYMENT_TARGET", "9.0");

**Comment by yasirkula** [comment]
It just feels like my `-framework` additions are either removed by some other post-processing script or ignored by Xcode for unknown reasons because I can see a variety of unknown symbols that come from all of these frameworks. You can try adding those frameworks via [PBXProject.AddFrameworkToProject](https://docs.unity3d.com/ScriptReference/iOS.Xcode.PBXProject.AddFrameworkToProject.html), if you wish. You should then comment out my AddBuildProperty lines.

**Comment by FrankdenUijl** [comment]
Hmm, like this?

```
Debug.Log("NGPostProcessBuild OnPostprocessBuild");
// Minimum supported iOS version on Unity 2018.1 and later is 8.0
#if !UNITY_2018_1_OR_NEWER
if( !Settings.Instance.MinimumiOSTarget8OrAbove )
{
	pbxProject.AddBuildProperty( targetGUID, "OTHER_LDFLAGS", "-weak_framework Photos" );
	pbxProject.AddBuildProperty( targetGUID, "OTHER_LDFLAGS", "-weak_framework PhotosUI" );
	pbxProject.AddBuildProperty( targetGUID, "OTHER_LDFLAGS", "-framework AssetsLibrary" );
	pbxProject.AddBuildProperty( targetGUID, "OTHER_LDFLAGS", "-framework MobileCoreServices" );
	pbxProject.AddBuildProperty( targetGUID, "OTHER_LDFLAGS", "-framework ImageIO" );
}
else
#endif
{
	pbxProject.AddFrameworkToProject( targetGUID, "PhotosUI.framework", true );
	pbxProject.AddFrameworkToProject( targetGUID, "Photos.framework", false );
	pbxProject.AddFrameworkToProject( targetGUID, "MobileCoreServices.framework", false );
	pbxProject.AddFrameworkToProject( targetGUID, "ImageIO.framework", false );
}

//pbxProject.RemoveFrameworkFromProject( targetGUID, "Photos.framework" );
```

**Comment by yasirkula** [comment]
Yeah that looks fine to me.

**Comment by FrankdenUijl** [comment]
Oke, thanks! I will let you know if it works.

**Comment by FrankdenUijl** [comment]
Hmm, I just checked same error.

The strange thing is that the OnPostprocessBuild does not get called at all.. I dont see the `NGPostProcessBuild OnPostprocessBuild` in the log on Unity Cloud Build. (I added that log before building)

That explain the error. Do you have any idea why?

**Comment by yasirkula** [comment]
Is it possible that **Settings.Instance.AutomatedSetup** returns false on Cloud Build? Does deleting [#if UNITY_IOS](https://github.com/yasirkula/UnityNativeGallery/blob/2cd01a22451ea2185a55a044c0e5e2bea96ca147/Plugins/NativeGallery/Editor/NGPostProcessBuild.cs#L91) allow OnPostprocessBuild to be called?

**Comment by FrankdenUijl** [comment]
I have just tested that it runs fine locally. Also found out that none of my "OnPostprocessBuild" methods are being called in Unity Cloud Build. 

![image](https://user-images.githubusercontent.com/8240783/144400824-75cc30e3-bfc9-4b3a-8b96-365aa9ad7871.png)
![image](https://user-images.githubusercontent.com/8240783/144400858-4e79466f-3c0f-41a3-add4-24b5b49b46d2.png)

But none of the logs shows up in UCB.

I have reported this to Unity.

**Comment by yasirkula** [comment]
How about you throw an exception there and see if it breaks the cloud build? It's so that we can eliminate the possibility of PostProcessBuild logs intentionally being omitted from Cloud Build by Unity.

**Comment by FrankdenUijl** [comment]
Nope, no exception, error or warning.

![image](https://user-images.githubusercontent.com/8240783/144419723-cc0bb3d3-a703-4943-b4cd-a8a14d120d9f.png)

**Comment by yasirkula** [comment]
Yeah, seems like a UCB issue then.

**Comment by FrankdenUijl** [comment]
I have fixed it by calling your method via a [Post-export method](https://docs.unity3d.com/Manual/UnityCloudBuildPreAndPostExportMethods.html). 

Seems like this issue is only in m project.

**Comment by yasirkula** [comment]
I'm curious, how did you figure out that this issue is only in your project?

**Comment by FrankdenUijl** [comment]
I run an old commit via UCB, it had your package in it (an older version) and other postprocessbuild methods. All of the post methods where invoked correctly. I only updated your package after I saw that UCB failed.

So something in my newer commits prevents post processing methods in UCB.

**Comment by yasirkula** [comment]
Did you possibly add a class named PostProcessBuildAttribute which has overridden Unity's own attribute?

**Comment by FrankdenUijl** [comment]
![image](https://user-images.githubusercontent.com/8240783/144837711-f460d422-0172-44b0-92ef-6da57d506605.png)
The only strange way of invoking OnPostprocessBuild.

But the package is excluded in our project.

**Comment by yasirkula** [comment]
I don't think you should investigate that ~ symbol. If you type PostProcessBuild and hover your cursor over it, make sure that it shows the namespace as `UnityEditor.Callbacks.PostProcessBuildAttribute`. Otherwise you have a conflicting custom PostProcessBuildAttribute. If the namespace is correct, I don't know how you can investigate this issue :D
